lib/charms/ovn_charm.py

```diff
@@ -283,6 +283,13 @@ def install(self):
 
         super().install()
 
+        # We need to render /etc/default/openvswitch-switch after the initial
+        # install and restart openvswitch-switch. This is done to ensure that
+        # when the disable-mlockall config option is unset, mlockall is
+        # disabled when running in a container.
+        self.render_configs(['/etc/default/openvswitch-switch'])
+        ch_core.host.service_restart('openvswitch-switch')
```

Comment:
This breaks the idempotency of the method, and it is called at every hook execution:
https://github.com/openstack-charmers/charm-layer-ovn/blob/b5a86ead9e130880a8a9a8c6c5cc861aeef3e9eb/reactive/ovn_chassis_charm_handlers.py#L72
https://github.com/openstack-charmers/charm-layer-ovn/blob/b5a86ead9e130880a8a9a8c6c5cc861aeef3e9eb/reactive/ovn_chassis_charm_handlers.py#L93

Normally calling it would be a no-op.

To do something on first install I'd check for the non-presence of the 'charm.installed' flag in reactive/ovn_chassis_charm_handlers.py and call some method from the charm class to handle it.

Reply:
Ah, bad assumption that it was only called once. I'll get that fixed.

Reply:
This is fixed, thanks for the review. Apologies for the forced push.

Reply:
Ty and no issue at all, I prefer the force-push approach as it gives a much cleaner commit history. If only GitHub would view the previous incarnations somewhere just like we see patch sets on Gerrit, now that would be something :-)